lua/conform/formatters/darker.lua

```diff
@@ -0,0 +1,14 @@
+local util = require("conform.util")
+---@type conform.FileFormatterConfig
+return {
+  meta = {
+    url = "https://github.com/akaihola/darker",
+    description = "Run black only on changed lines.",
+  },
+  command = "darker",
+  args = { "--stdout", "--quiet", "$FILENAME" },
```

Comment:
I think you're going to want to use `--stdin-filename`

Reply:
Huh. That's true. The weird thing is that I've been using `darker` first with `null-ls` and now with `conform.nvim` as a custom engine/formatter for quite a while now with those args and it's been working fine. Maybe by accident 😅 . Will fix!

Reply:
Hmm, I'm actually not sure if we want to use `--stdin-filename` here. Without it `darker` respects the revision option set in pyproject.toml if it's there so this way files formatted with `darker` via `conform` would match the files formatted with `darker .` in project root. Even if the root isn't a git repo.

Reply:
The trouble is what happens if you run this formatter on a buffer that has unsaved changes? It will read the current state from disk, which will blow away your changes. I assume you're using this formatter in a `BufWritePost` hook?

What do you think about doing something like this:
```lua
args = function(ctx)
  if vim.bo[ctx.buf].modified then
    return { "--stdout", "--quiet", "--stdin-filename", "$FILENAME" }
  else
    return { "--stdout", "--quiet", "$FILENAME" }
  end
end
```

Reply:
Oh yeah, I hadn't realized that auto-save.nvim had saved me from losing changes even when I had the formatter configured as `format_on_save`. Yeah, context aware args sound like a best-of-both-worlds solution to me. I'll update the PR shortly. Thank you for your patience and teaching me something new ^^